[LUA] Fix Geocolure spells shouldn't be castable with GEO sub

ampitere opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Prevents players who are sub GEO from casting Geocolure spells.

From Xaver: 
> went to retail to check. It seems #shitwikisais strikes again:
> 
> Main-job GEO can use Geo-spells with no handbell.
> Sub-job GEO can't because the spell doesn't even appear on the spell list at all.

## Steps to test these changes

1. !changejob 1 99
2. !changesjob 21 99
3. Attempt to cast a Geocolure spell either through text or a macro (it won't display them in the magic menu)

![image](https://github.com/LandSandBoat/server/assets/166072302/d4feb67a-19be-4a64-a8f7-0d16c50aa81d)

ampitere committed:
fix: Geocolure spells shouldn't be castable with GEO sub

Xaver-DaRed commented:
This fix is incorrect.
The requisite is to have a handbell equiped.

ampitere commented:
> This fix is incorrect. The requisite is to have a handbell equiped.

Good call, changed to check for handbell instead.

![image](https://github.com/LandSandBoat/server/assets/166072302/411ae51e-6d42-479e-9d93-0b4da29f166f)

Xaver-DaRed commented:
mmm.

I went to retail to check. It seems #shitwikisais strikes again:
- Main-job GEO can use Geo-spells with no handbell.
- Sub-job GEO can't because the spell doesn't even appear on the spell list at all.

So, sorry I missinformed you... However, I guess the job check WILL need a TODO to remove the Geo-spells from the spell list aswell.

So please:
- Change it so it checks for main job again. I guess the message shouldnt happen, but that one is fine aswell.
- Add a comment stating wikis are worng and no handbell is needd.
- Add a TODO to remove the geo-spells from the spell list IF they appear in the list

ampitere commented:
> mmm.
> 
> I went to retail to check. It seems #shitwikisais strikes again:
> 
> * Main-job GEO can use Geo-spells with no handbell.
> * Sub-job GEO can't because the spell doesn't even appear on the spell list at all.
> 
> So, sorry I missinformed you... However, I guess the job check WILL need a TODO to remove the Geo-spells from the spell list aswell.
> 
> So please:
> 
> * Change it so it checks for main job again. I guess the message shouldnt happen, but that one is fine aswell.
> * Add a comment stating wikis are worng and no handbell is needd.
> * Add a TODO to remove the geo-spells from the spell list IF they appear in the list

Reverted back to check for main GEO. Added a comment, TODO not necessary as it already does not appear in the spell list.
